dojo_theme/templates/module.html

```diff
@@ -64,20 +64,34 @@ <h4 class="accordion-item-name">{{ resource.name }}</h4>
   <h2>Challenges</h2>
 
   <div class="accordion" id="challenges">
-    {% for challenge in challenges %}
+    {% for i in range(0, challenges|length) %}
+      {% set challenge = challenges[i] %}
+
       {% set solved = "challenge-solved" if challenge.challenge_id in user_solves else "challenge-unsolved" %}
       {% set active = "challenge-active" if challenge.challenge_id == current_dojo_challenge.challenge_id else "" %}
-      {% call(header) accordion_item("challenges", loop.index) %}
+      {% set previous_challenge = challenges[i - 1] %}
+      {% set locked = challenge.visible() and challenge.data['progression_locked'] and previous_challenge.challenge_id not in user_solves and solved == "challenge-unsolved" %}
```

Comment:
You should be able to just do `challenge.progression_locked` (data_fields work like this).

Reply:
Let's also just special case the first challenge to not be locked even if it's progression_locked. Probably `loop.first` will be useful here.